CMakeLists.txt

```diff
@@ -50,6 +50,7 @@ target_link_libraries(zephyr_interface INTERFACE ${TOOLCHAIN_LIBS})
 # built purely with the include paths, defines, and other compiler
 # flags that come with zephyr_interface.
 zephyr_library_named(zephyr)
+target_link_libraries(zephyr kernel)
```

Comment:
It seems artificial to link 'zephyr' with 'kernel'. Why the library 'zephyr' and not the library 'app'?

And why just one Zephyr CMake library ('zephyr') and not all of them?

perhaps

zephyr_link_libraries(kernel)

would be more appropriate.